dashboard/lib/services/lti.rb

```diff
@@ -24,6 +24,9 @@ def self.initialize_lti_user(id_token)
       email: get_claim(id_token, :email),
     )
     user.authentication_options = [ao]
+
+    # As final step of the LTI user creation, a LtiUserIdentity is created for the user.
+    create_lti_user_identity(user)
```

Comment:
This won't work here, because `user` hasn't been created yet, it's only been "newed up", I.e. we haven't called `user.save`. That doesn't happen until the finish_signup is completed, and is handled in another controller.

If we do want to add the user identity here, we would need to do something similar what we are doing with the authentication options above, including adding [accepts_nested_attributes_for](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L221) to `user.rb`.

Reply:
I also think we'll need to add the nested lti_user_identies_attributes to the object we're returning here. I need to look deeper into this to remember how this all works https://github.com/code-dot-org/code-dot-org/pull/54374/files#diff-d00fda1c5d36968c46dc34d1f192623e04c3e415612db0570917dbefb2cde443R1

Reply:
Cool, the idea was to have a visual of what the desired behaviour is. I'll look into this pointer you gave me.

Reply:
I´m leaving a TODO note instead of including those changes in this PR. I consider an update in the user model complex enough to have it's own PR in case a revert is needed and for documentation purposes.